common/beerocks/bcl/include/bcl/beerocks_event_loop.h

```diff
@@ -0,0 +1,157 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BEEROCKS_EVENT_LOOP_H_
+#define _BEEROCKS_EVENT_LOOP_H_
+
+#include <functional>
+#include <memory>
+
+// Forward declaration
+class Socket;
```

Comment:
The `Socket` class is currently used since it's the only file descriptor wrapper we currently have, and the event loop will be used with sockets anyway... We could create an `FD` class and use it instead, but I'm not sure what is the added value, as in most cases we'll have to cast `FD` to `Socket` anyway...

Reply:
At least, we are going to use EventLoop with timers too.
I agree with you on that creating an `FD` as suggested is not worth it. 
There must be other ways to look up the Socket from its file descriptor in the event handler.

Reply:
I think the template is the way to go here.

However, I think we can change this later. I'd prefer to get this PR merged ASAP and extend/improve from there.

Reply:
Updated to a templated based implementation: https://github.com/prplfoundation/prplMesh/pull/1435/commits/db9df3d8addefd05d224da891fa957cee9365e8b (fixup)